compiler/passes/insertLineNumbers.cpp

```diff
@@ -153,15 +152,11 @@ insertLineNumber(CallExpr* call) {
   } else if (fn->hasFlag(FLAG_EXTERN)                           ||
              (fn->hasFlag(FLAG_EXPORT) &&
               !fn->hasFlag(FLAG_INSERT_LINE_FILE_INFO))         ||
-             strcmp(fn->name, "chpl__heapAllocateGlobals") == 0 ||
-             strcmp(fn->name, "chpl__initStringLiterals")  == 0 ||
-             strcmp(fn->name, "chpl__initModuleGuards")    == 0 ||
-             strcmp(fn->name, "chpl_gen_main")             == 0 ||
```

Comment:
We changed these removed functions into an assert, which has since been removed. I'd recommend leaving the assert in for this PR, with a TODO note that it should be removed. In any case it needs to pass full local testing with the assert on. Also AFAIK heapAllocateGlobals might only be relevant in GASNet configurations. Hence the assert to catch errors in our logic with nightly testing.